hathor/p2p/netfilter/rule.py

```diff
@@ -28,6 +28,32 @@ def __init__(self, match: 'NetfilterMatch', target: 'NetfilterTarget'):
         self.match = match
         self.target = target
 
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, NetfilterRule):
+            return NotImplemented
+
+        self_chain_name = self.chain.name if self.chain else None
```

Comment:
I don't feel it's worth accessing attributes of the chain object. Why don't you just do `self.chain == other.chain`?

Reply:
This code does not exist anymore